src/datachain/lib/signal_schema.py

```diff
@@ -282,7 +299,18 @@ def clone_without_file_signals(self) -> "SignalSchema":
         return SignalSchema(schema)
 
     def mutate(self, args_map: dict) -> "SignalSchema":
-        return SignalSchema(self.values | sql_to_python(args_map))
+        new_values = self.values.copy()
+
+        for name, value in args_map.items():
+            if isinstance(value, Column) and value.name in self.values:
+                # renaming existing signal
+                del new_values[value.name]
```

Comment:
misc: del is not needed becasue of the next line

Reply:
It's needed. Old signals is removed and new one added. It's different keys